applications/graphQL/query.py

```
@@ -0,0 +1,96 @@
+    )
+
+    def resolve_random_insult(
+        root, info, category: Optional[str] = None
+    ) -> Optional[Insult]:
+        """Resolve a random active insult, optionally filtered by category."""
+        queryset = Insult.objects.filter(status="A")
+        if category:
+            queryset = queryset.filter(category=category)
+        return queryset.order_by("?").first()
+
+    def resolve_insult_by_id(root, info, id: str) -> Insult:
```

Ruff (0.12.2)

Missing type annotation for function argument `info`

(ANN001)